hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileWriterImpl.java

```diff
@@ -798,16 +809,15 @@ protected void finishFileInfo() throws IOException {
     }
 
     // Average key length.
-    int avgKeyLen =
-        entryCount == 0 ? 0 : (int) (totalKeyLength / entryCount);
+    int avgKeyLen = (int)this.keySizeSketch.getQuantile(0.5);
```

Comment:
Please call this median and not average

Reply:
No problem. There are AVG_ defines in HFileInfo that I haven't touched.